src/main/kotlin/org/rust/ide/inspections/RsDetachedFileInspection.kt

```diff
@@ -62,6 +62,12 @@ class RsDetachedFileInspection : RsLocalInspectionTool() {
         return null
     }
 
+    override val requirements: Set<InspectionRequirement> = setOf(
+        InspectionRequirement.Toolchain,
```

Comment:
Is it needed to distinguish between project and workspace here, i.e. is there any inspection that needs that distinction?

Reply:
That's a good question, I'm not sure about this separation too and just left the most general implementation for this review. Let's wait for a review from @vlad20012

Reply:
I agree with @Kobzol and think that we should merge `Project` and `Workspace` because I hardly can come up with situation when these requirements may be different

Reply:
Thanks, I got rid of `Project` requirement